File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.Utils;


namespace DXSample {
    public partial class Main: XtraForm {
        public Main() {
            InitializeComponent();
        }
        public void InitData() {
            for(int i = 0;i < 5;i++) 
                dataSet11.Tables[0].Rows.Add(new object[] { i, string.Format("FirstName {0}", i), i,  DateTime.Today.AddDays(i), true });
            
        }
        SortDialogProvider provider;
        private void OnLoad(object sender, EventArgs e) {
            InitData();
            provider = new SortDialogProvider(gridView1);
            provider.EnableSortDialog();
           // provider.ShowSortIndexInColumnHeader 
        }

        private void OmFormClosing(object sender, FormClosingEventArgs e)
        {
            provider.DisableSortDialog();
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.Utils


Namespace DXSample
	Partial Public Class Main
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
		End Sub
		Public Sub InitData()
			For i As Integer = 0 To 4
				dataSet11.Tables(0).Rows.Add(New Object() { i, String.Format("FirstName {0}", i), i, Date.Today.AddDays(i), True })
			Next i

		End Sub
		Private provider As SortDialogProvider
		Private Overloads Sub OnLoad(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			InitData()
			provider = New SortDialogProvider(gridView1)
			provider.EnableSortDialog()
		   ' provider.ShowSortIndexInColumnHeader 
		End Sub

		Private Sub OmFormClosing(ByVal sender As Object, ByVal e As FormClosingEventArgs) Handles Me.FormClosing
			provider.DisableSortDialog()
		End Sub
	End Class
End Namespace
